### ISSUE TITLE
nix copy doesn't work for drv files

### ISSUE THREAD

**Issue by LnL7** [NixOS/nix#1639]
With nix-copy-closure it's possible to build a drv on a remote machine without using the build hook. This is useful to avoid substituting dependencies over a slow connection.

```
$ nix copy --to ssh://desktop /path/to/build.drv
$ ssh desktop -- nix build /path/to/build.drv
cannot build missing derivation '/path/to/build.drv'
```

**Comment by volth** [comment]
it seems that ```nix copy --to ssh:// ...``` uploads all files in a closure (.patch, .sh, ...) except .drv

```nix-copy-closure``` works well but much slower :(

**Comment by edolstra** [comment]
I haven't really decided whether the `nix` should support .drv files at all, since we might want to remove them at some point.

**Comment by edolstra** [comment]
It's not quite clear what a drv-less future would look like, but probably you would do something like `nix build --store ssh://bla nixpkgs.hello`.

**Comment by volth** [comment]
> but probably you would do something like ```nix build --store ssh://bla nixpkgs.hello.```

That's exactly what I am trying to avoid.
We already have a magic way (```nix-build --builders```) which executes half of the builds locally and abort the build on network failures.

I am talking about 4 steps which can be executed by an external scheduler.
1. prepare build instructions (```nix-instantiate```)
2. deploy build instructions (```nix-copy-closure --to```)
3. do the build (```nix-store -r```)
3. collect results of the build  (```nix-copy-closure --from```)

Do you mean that soon it won't be possible to decouple the distributed build into the steps?

**Comment by oxij** [comment]
This is an issue for me too since I do

> 1. prepare build instructions (```nix-instantiate```)
> 2. deploy build instructions (```nix-copy-closure --to```)
> 3. do the build (```nix-store -r```)
> 3. collect results of the build  (```nix-copy-closure --from```)

from automation scripts (I posted on of them before to nix-dev ML) a lot.

**Comment by edolstra** [comment]
Fixed in 85530bcc0b4c61499f90fa6125c19beeb0f4485e.
